packages/core/strapi/lib/services/event-hub.js

```diff
@@ -1,16 +1,78 @@
+'use strict';
+
 /**
  * The event hub is Strapi's event control center.
  */
+module.exports = function createEventHub() {
+  const listeners = new Map();
 
-'use strict';
+  // Default subscriber to easily add listeners with the on() method
+  const defaultSubscriber = async (eventName, ...args) => {
+    if (listeners.has(eventName)) {
+      for (const listener of listeners.get(eventName)) {
+        await listener(...args);
+      }
+    }
+  };
 
-const EventEmitter = require('events');
+  // Store of subscribers that will be called when an event is emitted
+  const subscribers = [defaultSubscriber];
 
-class EventHub extends EventEmitter {}
+  const eventHub = {
+    async emit(eventName, ...args) {
+      for (const subscriber of subscribers) {
+        await subscriber(eventName, ...args);
+      }
+    },
 
-/**
- * Expose a factory function instead of the class
- */
-module.exports = function createEventHub(opts) {
-  return new EventHub(opts);
+    subscribe(subscriber) {
+      subscribers.push(subscriber);
+
+      // Return a function to remove the subscriber
+      return () => {
+        eventHub.unsubscribe(subscriber);
+      };
+    },
+
+    unsubscribe(subscriber) {
+      subscribers.splice(subscribers.indexOf(subscriber), 1);
+    },
+
+    on(eventName, listener) {
+      if (!listeners.has(eventName)) {
+        listeners.set(eventName, []);
+      }
+
+      listeners.get(eventName).push(listener);
```

Comment:
I know you answered this on the other PR but I'm curious what you think about an if else block like below, or why you decided against it.

```suggestion
      if (!listeners.has(eventName)) {
        listeners.set(eventName, [listener]);
      } else {
        listeners.get(eventName).push(listener);
      }


```

Reply:
Yes I do think that's better, we'll need to open a small pr for it

Reply:
done in #15262